pkg/ext-proc/backend/provider.go

```diff
@@ -217,3 +227,40 @@ func (p *Provider) refreshMetricsOnce() error {
 	}
 	return errs
 }
+
+func (p *Provider) flushPrometheusMetricsOnce() {
+	klog.V(logutil.DEFAULT).Infof("Flushing Prometheus Metrics")
+
+	pool, _ := p.datastore.getInferencePool()
+	if pool == nil {
+		// No inference pool or not initialize.
+		return
+	}
+
+	var kvCacheTotal float64
+	var queueTotal int
+
+	podTotalCount := 0
+	pods, err := p.datastore.getPods()
+	if err != nil {
+		klog.V(logutil.DEFAULT).Infof("Couldn't list pods: %v", err)
+		return
+	}
+
+	for _, pod := range pods {
+		if !podIsReady(pod) {
+			continue
+		}
+		podTotalCount++
+		if val, ok := p.podMetrics.Load(pod.Name); ok {
+			pm := val.(*PodMetrics)
+			kvCacheTotal += pm.KVCacheUsagePercent
```

Comment:
Just a high level thought....

As an optimization, would we ever consider doing this calculation as part of the actual logic in https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/3ff0af85f9341468c6f4b1e2923dd8f9b413b7e2/pkg/ext-proc/scheduling/filter.go#L134?

Then we are computing these metrics directly in-line with the endpoint picking logic and could get the absolute freshest value.

Reply:
Actually, one thing to consider here is that the probing will be frequent, and currently the podMetrics map only reflects the latest probed value. We should consider aggregating over a time window to avoid oscillations. @liu-cong @kaushikmitr  did we think about this in the context of the endpoint picking algorithm (i.e., using the absolute last value vs aggregation over a window)?